modules/drivers/snowflake/test/metabase/driver/snowflake_test.clj

```diff
@@ -166,22 +190,51 @@
           (is (= expected
                  (driver/describe-database :snowflake (assoc (mt/db) :name "ABC")))))))))
 
+(deftest describe-database-default-schema-test
+  (testing "describe-database should include Tables from all schemas even if the DB has a default schema (#38135)"
+    (mt/test-driver :snowflake
+      (let [details     (assoc (mt/dbdef->connection-details :snowflake :db {:database-name "lowercase_test"})
+                               ;; simulate a DB default schema or session schema by including it in the connection
+                               ;; details... see
+                               ;; https://metaboat.slack.com/archives/C04DN5VRQM6/p1706219065462619?thread_ts=1706156558.940489&cid=C04DN5VRQM6
+                               :schema "PUBLIC"
+                               :schema-filters-type "inclusion"
+                               :schema-filters-patterns "test_schema")
+            db-name     (:db details)
+            schema-name "test_schema"
+            table-name  "test_table"
+            field-name  "test_id"
+            spec        (sql-jdbc.conn/connection-details->spec :snowflake details)
+            identifier  (fn [& args]
+                          (str/join \. (map #(str \" % \") args)))]
+        ;; create the snowflake DB
+        (doseq [stmt [(format "DROP DATABASE IF EXISTS %s;" (identifier db-name))
+                      (format "CREATE DATABASE %s;" (identifier db-name))
+                      (format "CREATE SCHEMA %s;" (identifier db-name schema-name))
+                      (format "CREATE TABLE %s (%s INTEGER AUTOINCREMENT);" (identifier db-name schema-name table-name) (identifier field-name))
+                      (format "GRANT SELECT ON %s TO PUBLIC;" (identifier db-name schema-name table-name))]]
+          (jdbc/execute! spec [stmt] {:transaction? false}))
+        ;; create the DB object
+        (t2.with-temp/with-temp [Database database {:engine :snowflake, :details details}]
+          (is (=? {:tables #{{:name table-name, :schema schema-name, :description nil}}}
+                  (driver/describe-database :snowflake database))))))))
+
 (deftest describe-database-views-test
   (mt/test-driver :snowflake
     (testing "describe-database views"
       (let [details (mt/dbdef->connection-details :snowflake :db {:database-name "views_test"})
+            db-name (:db details)
             spec    (sql-jdbc.conn/connection-details->spec :snowflake details)]
         ;; create the snowflake DB
-        (jdbc/execute! spec ["DROP DATABASE IF EXISTS \"views_test\";"]
```

Comment:
Also unrelated, I realized this test is dropping and creating the database `views_test` every time it runs and not prefixing it for the current test session at all... so if two runs were to run this same exact test at the same time it would cause one of them to flake. Maybe this hasn't happened yet, but I fixed it either way to be safe and to set a good example for other people writing tests in the future